server/player/handler.go

```diff
@@ -22,6 +22,8 @@ type Handler interface {
 	HandleMove(ctx *event.Context, newPos mgl64.Vec3, newYaw, newPitch float64)
 	// HandleTeleport handles the teleportation of a player. ctx.Cancel() may be called to cancel it.
 	HandleTeleport(ctx *event.Context, pos mgl64.Vec3)
+	// HandleChangeWorld handles when the player is added to a new world. oldWorld may be nil.
+	HandleChangeWorld(oldWorld *world.World, newWorld *world.World)
```

Comment:
Would go back to calling this just `old` and `new`, the types make it clear these are worlds. Should be good to go after.

Reply:
new conflicts with Go's `new` function

Reply:
Could name them `before` and `after` then.